src/mavlink/utils.rs

```diff
@@ -0,0 +1,129 @@
+use anyhow::anyhow;
+use tracing::*;
+
+use super::mavlink_camera_component::MavlinkCameraComponent;
+
+pub fn from_string_to_u8_array_with_size_32(src: &str) -> [u8; 32] {
+    let bytes = src.as_bytes();
+    let mut dst = [0u8; 32];
+    let len = std::cmp::min(bytes.len(), 32);
+    dst[..len].copy_from_slice(&bytes[..len]);
+    dst
+}
+
+pub fn from_string_to_char_array_with_size_32(src: &str) -> [char; 32] {
+    let chars: Vec<char> = src.chars().collect();
+    let mut dst = ['\0'; 32];
+    let len = std::cmp::min(chars.len(), 32);
+    dst[..len].copy_from_slice(&chars[..len]);
+    dst
+}
+
+pub fn from_string_to_vec_char_with_defined_size_and_null_terminator(
+    src: &str,
+    size: usize,
+) -> Vec<char> {
+    let mut uri = src.chars().collect::<Vec<char>>();
+    uri.truncate(size);
+    uri.push('\0');
+    uri
+}
```

Comment:
I was thinking about it and.. should it be:
```suggestion
pub fn from_string_to_vec_char_with_defined_size_and_null_terminator(
    src: &str,
    size: usize,
) -> Vec<char> {
    let mut uri = src.chars().collect::<Vec<char>>();
    uri.truncate(size - 1);
    uri.push('\0');
    uri
}
```

Reply:
The truncate does not make much sense to me. we should just add \0

Reply:
> The truncate does not make much sense to me. we should just add \0

The truncation is technical here: The protocol states, for example, 140 characters long, so our vector needs to have, at maximum, 140 characters long. We could create the vector with a capacity of 140 characters and copy each character into it, or to have this unsized Vec and then truncate it to the correct size.

Reply:
> > The truncate does not make much sense to me. we should just add \0
> 
> The truncation is technical here: The protocol states, for example, 140 characters long for a field, so our vector needs to have, at maximum, 140 characters long. We could create the vector with a capacity of 140 characters and copy each character into it, or, as it is being done here, to have this unsized Vec and then truncate it to the correct size (size of the original string, truncated at 140) after copying the characters into it.

Oh sorry, I thought that was str.size and not size as parameter. It makes totally sense. Truncate will limit the size of the array. To be safer, size -1 makes more sense indeed.

Reply:
Cool, instead of `truncate(lenght-1)`, I changed it to use `take(lenght-1)`.